test/jdk/java/awt/event/ComponentEvent/ListItemEventsTest.java

```diff
@@ -0,0 +1,193 @@
+
+    public static void initializeGUI() {
+        frame = new Frame("Test Frame");
+        frame.setLayout(new FlowLayout());
+        list = new List();
+        list.add("One");
+        list.add("Two");
+        list.add("Three");
+        list.add("Four");
+        list.add("Five");
+        list.addItemListener((event) -> {
+            System.out.println("Got an ItemEvent:" + event);
+            itemStateChanged = true;
+        });
+        list.addActionListener((event) -> {
+            System.out.println("Got an ActionEvent:" + event);
+            actionPerformed = true;
+        });
+        list.addFocusListener(new FocusAdapter() {
+            public void focusGained(FocusEvent event) {
+                System.out.println("Got an FocusEvent:" + event);
+                latch.countDown();
+            }
+        });
+
+        frame.add(list);
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    public static void main(String[] s) throws Exception {
+        latch = new CountDownLatch(1);
+        robot = new Robot();
+        try {
+
+            robot.setAutoDelay(100);
+            robot.setAutoWaitForIdle(true);
+
+            EventQueue.invokeLater(ListItemEventsTest::initializeGUI);
+            robot.waitForIdle();
+
+            Point listAt = list.getLocationOnScreen();
+            // get bounds of button
+            Rectangle bounds = list.getBounds();
+
+            robot.mouseMove(listAt.x + bounds.width / 2,
+                listAt.y + bounds.height / 2);
+
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+            if (!latch.await(15, TimeUnit.SECONDS)) {
+                throw new RuntimeException(
+                    "Fail: Timed out waiting for list to gain focus, test cannot proceed!!");
+            }
+
+            if (!itemStateChanged) {
+                throw new RuntimeException(
+                    "FAIL: List did not trigger ItemEvent when item selected!");
+            }
+
+            robot.mousePress(MouseEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(MouseEvent.BUTTON1_DOWN_MASK);
+            robot.mousePress(MouseEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(MouseEvent.BUTTON1_DOWN_MASK);
+
+            if (!actionPerformed) {
+                throw new RuntimeException(
+                    "FAIL: List did not trigger ActionEvent when double"
+                        + " clicked!");
+            }
+
+            itemStateChanged = false;
+            actionPerformed = false;
+
+            EventQueue.invokeAndWait(() -> list.select(0));
+
+            if (itemStateChanged) {
+                throw new RuntimeException(
+                    "FAIL: List triggered ItemEvent when item selected by "
+                        + "calling the API!");
+            }
+
+            robot.setAutoDelay(waitDelay);
```

Comment:
Why is delay increased here?

Reply:
This test verifies list section via mouse/keys events.
 
For mouse events, robot.setAutoDelay(100) worked well. But for key events there were a few intermittent failures which  cleared out with a higher delay via robot.setAutoDelay(1000)

Reply:
I see. Would it be clearer if you have two constants:
```java
private static final int MOUSE_DELAY = 100;
private static final int KEYBOARD_DELAY = 1000;
```

Then you would set the former initially and, before testing with key events, you would change to the latter. It'll make the intention much clearer and explain why it's changed.

Reply:
Resolved . implemented as suggested.